### PR TITLE
add secondsmicro option for unixtrunc builtin

### PR DESCRIPTION
**Pull request by bpopadiuk**
adds an option `"secondsmicro"` to `unixtrunc()` to output unix seconds with microsecond precision after the decimal:

```
    unixtrunc("1438445529707123456", "secondsmicro") --> "1438445529.707123"
    unixtrunc("2020-05-08 19:00:00.999998", secondsmicro) --> "1588964400.999998"
```

### CODE REVIEW

File: expr/builtins/time.go [hunk 1]
```diff
@@ -528,18 +529,21 @@ func timeTruncEvalOne(ctx expr.EvalContext, args []value.Value) (value.Value, bo
...
+		milliTs = itemT.Int()
 	default:
 		// If not a TimeValue, convert to a TimeValue and get Unix w/ milliseconds.
 		t, ok := value.ValueToTime(args[0])
 		if !ok || t.IsZero() {
 			return value.NewStringValue(""), false
 		}
-		valTs = t.UnixNano() / 1e6
+		microTs = t.UnixNano() / 1e3
+		milliTs = microTs / 1e3
```
Comment (trace-andreason) [inline expr/builtins/time.go:546]: I see what you are doing here, but I think it's a little confusing. Might just be me though

Comment (bpopadiuk) [reply expr/builtins/time.go:546]: yeah, do you think deriving both from `t.UnixNano()` would be clearer?

Comment (trace-andreason) [reply expr/builtins/time.go:546]: I think my confusion was mainly looking at the diff and seeing you used a different number. I think its fine

**Comment by junichif** [comment]
@araddon can you take a look at this PR and see if it looks ok?

**Pull request merged**